src/app/3d/qgsphongmaterialwidget.cpp

```diff
@@ -0,0 +1,35 @@
+#include "qgsphongmaterialwidget.h"
+
+#include "qgsphongmaterialsettings.h"
+
+
+QgsPhongMaterialWidget::QgsPhongMaterialWidget( QWidget *parent )
+  : QWidget( parent )
+{
+  setupUi( this );
+
+  setMaterial( QgsPhongMaterialSettings() );
+
+  connect( btnDiffuse, &QgsColorButton::colorChanged, this, &QgsPhongMaterialWidget::changed );
```

Comment:
Can these colors have alpha? If so, you'll need to setAllowOpacity on the buttons

Reply:
No, alpha is currently not supported - transparency in 3D world is unfortunately not that straightforward :-)